### PR TITLE
Rooms

### PR DESCRIPTION
**Pull request by GreyMatteOr**
# Description

Added a new `Room` class. Each Room has it's own game and set of players. Implemented a basic Create Room / Join Room architecture.

See the wireframe for basic room flow.

# where to start

Full demo

# Notes: 
 - Still need to add a leave room functionality (other than refreshing)
 - Still need to test

# Issues
[Issue #56](https://github.com/GreyMatteOr/bangwords/issues/56)
[Corresponding UI PR](https://github.com/GreyMatteOr/bangwords/pull/58)

### CODE REVIEW

File: Room/Room.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+const Game = require('../Game/Game.js');
+
+class Room {
+  constructor( id ) {
+    this.id = id;
+    this.players = {};
+    game = new Game();
+  }
+
+  addPlayer( id, name ) {
```
Comment (holladayian) [inline Room/Room.js:10]: Lets use something like playerID and roomID or something for dev empathy

File: server.js [hunk 2]
```diff
@@ -13,6 +14,25 @@ app.set('port', process.env.PORT || 3001);
 io.on( "connect", ( socket ) => {
   console.log(`${socket.id.slice(0, -8)} connected.`)
 
+  socket.on( 'createRoom', ( id ) => {
+    if (!room[id]) {
+      room[id] = new Room(id);
+      io.to(socket.id).emit( 'joinRoom', id )
+    } else {
+      io.to(socket.id).emit('result', {errorMSG: `A room with the name '${id}' already exists! Choose again`});
```
Comment (holladayian) [inline server.js:23]: Just a note to display this later. We can put this in an issue

File: server.js [hunk 3]
```diff
@@ -13,48 +14,75 @@ app.set('port', process.env.PORT || 3001);
...
-    isGameReady: isGameReady(),
-    hasGenerator: game.generatorID !== null
+function cleanData(roomID, socket) {
+  socket.leave(roomID);
+  let room = rooms[roomID];
+  room.deletePlayer(socket.id);
+  delete players[socket.id];
+  if (room.getPlayerCount() <= 0) {
+    delete rooms[roomID];
   }
```
Comment (holladayian) [inline server.js:86]: I like this! Adds to the scalability ❤️

**Review by holladayian — APPROVED** [review]
Looks good! No major comments. A few things we can make issues out of if we need.

File: Room/Room.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+const Game = require('../Game/Game.js');
+
+class Room {
+  constructor( id ) {
+    this.id = id;
+    this.players = {};
+    this.game = new Game();
```
Comment (Chriscastanuela) [inline Room/Room.js:7]: Genius

**Review by Chriscastanuela — APPROVED** [review]

**Pull request merged**